From 0000000000000000000000000000000000000005 Mon Sep 17 00:00:00 2001
From: MC OS <mcos@local>
Date: Mon, 6 Jul 2026 00:00:00 +0000
Subject: [PATCH] neardi lb200: YT8521 PHY refclk, delay and motorcomm props

Enable REFCLKO25M (25 MHz PHY ref), program RX internal delay for
rgmii-rxid, and motorcomm sleep/PLL flags matching SDK.
---
 arch/arm64/boot/dts/rockchip/rk3576-neardi-lb200-gmac.dtsi | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3576-neardi-lb200-gmac.dtsi b/arch/arm64/boot/dts/rockchip/rk3576-neardi-lb200-gmac.dtsi
index 1111111..2222222 100644
--- a/arch/arm64/boot/dts/rockchip/rk3576-neardi-lb200-gmac.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3576-neardi-lb200-gmac.dtsi
@@ -48,6 +48,11 @@
 		/* YT8521 needs broadcast address 0x0 */
 		reg = <0x0>;
 		clocks = <&cru REFCLKO25M_GMAC0_OUT>;
+		assigned-clocks = <&cru REFCLKO25M_GMAC0_OUT>;
+		assigned-clock-rates = <25000000>;
+		rx-internal-delay-ps = <1950>;
+		motorcomm,auto-sleep-disabled;
+		motorcomm,keep-pll-enabled;
 	};
 };
 
@@ -57,5 +62,10 @@
 		/* YT8521 needs broadcast address 0x0 */
 		reg = <0x0>;
 		clocks = <&cru REFCLKO25M_GMAC1_OUT>;
+		assigned-clocks = <&cru REFCLKO25M_GMAC1_OUT>;
+		assigned-clock-rates = <25000000>;
+		rx-internal-delay-ps = <1950>;
+		motorcomm,auto-sleep-disabled;
+		motorcomm,keep-pll-enabled;
 	};
 };
-- 
2.34.1
